src/tests/ftest/telemetry/pool_svc_metrics.py

```diff
@@ -0,0 +1,133 @@
+'''
+  (C) Copyright 2024 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+'''
+import time
+
+from telemetry_test_base import TestWithTelemetry
+
+MAP_VERSION_METRIC = "engine_pool_svc_map_version"
+SVC_LEADER_METRIC = "engine_pool_svc_leader"
+DEGRADED_RANKS_METRIC = "engine_pool_svc_degraded_ranks"
+
+
+class PoolServiceMetrics(TestWithTelemetry):
+    """Verify pool service metric values.
+
+    :avocado: recursive
+    """
+
+    def collect_svc_telemetry(self, pool_uuid):
+        """Collect the pool service metric values.
+
+        Args:
+            pool_uuid: The UUID of the pool to collect metrics from.
+
+        Returns:
+            dict: A dict of the current pool service leader's metrics.
+        """
+        def _map_version(rank_metrics):
+            if rank_metrics is None or MAP_VERSION_METRIC not in rank_metrics:
+                return 0
+            return rank_metrics[MAP_VERSION_METRIC]
+
+        def _pool_rank_metrics(host_metrics, pool_uuid):
+            self.log.debug("collecting rank metrics for pool: %s", pool_uuid)
+            pm = {}
+            for hm in host_metrics.values():
+                for k, v in hm.items():
+                    if 'metrics' not in v:
+                        continue
+                    for m in v['metrics']:
+                        if 'labels' not in m:
+                            continue
+                        if 'pool' not in m['labels'] or \
+                                m['labels']['pool'].casefold() != str(pool_uuid).casefold():
+                            continue
+                        if 'rank' not in m['labels']:
+                            continue
+                        rank = m['labels']['rank']
```

Comment:
It's somewhat personal preference, but FYI you could do
```suggestion
                        try:
                            if m['labels']['pool'].casefold() != str(pool_uuid).casefold():
                                continue
                            rank = m['labels']['rank']
                        except KeyError:
                            continue
```

Reply:
Hmm, good point. I've gotten out of the Python way of thinking. That's cleaner.